### PR TITLE
Update versionize crate documentation.

### PR DESCRIPTION
**Pull request by sandreim**
## Reason for This PR

Update versionize crate documentation.

## Description of Changes

Provided in commit messages.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -2,26 +2,31 @@
...
-//! implements a persistent storage format for Firecracker state snapshots.
-//! The `Versionize` trait defines a generic interface that serializable state structures
-//! need to implement.
+//! Defines a generic interface for version tolerant serialization and
+//! implements it for primitive data types using `bincode` as backend.
 //!
-//! `VersionMap` exposes an API that maps the individual structure versions to
-//! a root version. This mapping is required both when serializing or deserializing structures as
-//! it needs to know which version of structure to serialize for a given target data version.
+//! The interface has two parts:
```
Comment (georgepisaltu) [inline src/lib.rs:8]: nit
```suggestion
//! The interface has two components:
```

File: src/version_map.rs [hunk 2]
```diff
@@ -1,35 +1,89 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-//! Maps struct/enum/union versions to a sequence of root versions.
-//! This is required to support the concept of a snapshot version
-//! composed of individually versioned components.
+//! A helper to map struct and enum versions to a sequence of root versions.
+//! This helper required to support the versioning of a hierarchy of structures
```
Comment (luminitavoicu) [inline src/version_map.rs:5]:
```suggestion
//! This helper is required to support the versioning of a hierarchy of structures
```

File: src/version_map.rs [hunk 3]
```diff
@@ -1,35 +1,89 @@
...
+//! composed of individually versioned structures or enums.
+//!
+//! In the example below, Struct
```
Comment (luminitavoicu) [inline src/version_map.rs:8]: This seems unfinished.

**Comment by sandreim** [comment]
@luminitavoicu @georgepisaltu updated, PTAL.

**Review by luminitavoicu — APPROVED** [review]

**Review by berciuliviu — APPROVED** [review]

**Review by georgepisaltu — APPROVED** [review]

**Review by berciuliviu — APPROVED** [review]

**Pull request merged**